### DISCUSSION TITLE
How to have different user/roles edit different portions of different pages?

### DISCUSSION THREAD

**Discussion by daughtkom** [OrchardCMS/OrchardCore#10032] [Q&A]
In non-Orchard terminology, I'm trying to have multiple content blocks (HTML block, list of links, etc.) each separately editable by one or more content creators (editors?).  Different people/roles would need access to manage different blocks of content.  Preferably role-based permissions instead of user-based.  Then I would have another user with different role(s) assemble these blocks into one or more CMS pages.

How can I do this in Orchard Core? I've done some development with Orchard Core about 18 months ago, but nothing like this.

While I want to make it easy for myself as the developer, I'm most interested in making it easy for those that would be managing the content (editing blocks, creating pages, assembling blocks into pages, etc.).  For example, the blocks could be edited individually as standalone content items, or they could be edited "in place" within a page as long as the permissions were appropriately locked down.  

Any ideas?  I'm sure this doesn't sound as clear in this post as the need is in my head, so let me know what additional info I can provide.

**Comment by Piedone** [comment]
I think what you're looking for is to have permissions for [widgets](https://docs.orchardcore.net/en/latest/docs/reference/modules/Widgets/) in some way (when talking about "content blocks" on a page, the Orchard feature to use is most possibly widgets).

Widgets are content items and as such can be configured with the Securable metadata (see each widget content type's editor) which in turn lets you configure which user role possesses which permissions corresponding to that widget content type. This allows you to let certain user roles edit Html widgets but not Paragraph widgets, for example.

Note that this is only about permissions for whole content types, e.g. the Html widget content type, meaning that it'll be applied to all Html widget content items. Finer-grained permissions on the content item level are not yet available (out of the box), see: https://github.com/OrchardCMS/OrchardCore/issues/5925
